### PR TITLE
Update iter.rs

### PR DESCRIPTION
**Pull request by kineticz**
error in example

### CODE REVIEW

**Comment by steveklabnik** [comment]
@bors: r+ rollup


Thanks so much!

File: src/libcore/iter.rs [hunk 1]
```diff
@@ -2853,7 +2853,7 @@ impl<'a, I: DoubleEndedIterator + ?Sized> DoubleEndedIterator for &'a mut I {
 /// impl ExactSizeIterator for Counter {
 ///     // We already have the number of iterations, so we can use it directly.
 ///     fn len(&self) -> usize {
-///         self.count
+///         5 - self.count
 ///     }
 /// }
 ///
```
Comment (bluss) [inline src/libcore/iter.rs]: can you fix the assertion below too? It should test that the len is 5.

Comment (kindlychung) [inline src/libcore/iter.rs]: Done.

**Comment by bluss** [comment]
Note that this PR is not mergeable.

**Comment by kindlychung** [comment]
What else can I do?

**Comment by bluss** [comment]
Rebase your branch so that it applies to current master.

**Comment by kindlychung** [comment]
Uh, not exactly sure how to do that. I did all the editing on github.

**Comment by steveklabnik** [comment]
@kindlychung you have to pull it down locally, github doesn't have a rebase in the web UI

**Comment by kindlychung** [comment]
Looks like the original file has been removed. The new file is iter/iterator.rs.

Please close this. I will file a new PR.

**Comment by steveklabnik** [comment]
Ah! Thanks.

**Pull request closed without merging**